### DISCUSSION TITLE
Minimal distance may not be sufficient to identify a failure

### DISCUSSION THREAD

**Discussion by dgumenyuk** [skhatiri/UAV-Testing-Competition#10] [Evaluation] [answered]
[results1.zip](https://github.com/skhatiri/UAV-Testing-Competition/files/13652059/results1.zip)
![test_7](https://github.com/skhatiri/UAV-Testing-Competition/assets/17391422/18417691-0727-4328-9a5c-f473bc22cd27)
Take a look at the following test. Evidently, it failed, however the minimal distance is more, than 1.5.
I attach the logs in the .zip. 
How will such test case be evaluated in the context of the competition?

**Comment by dgumenyuk** [comment]
I am providing another example, where the test seems to fail, but the distance is bigger, than  1.5:
![test_46](https://github.com/skhatiri/UAV-Testing-Competition/assets/17391422/51ff6936-3008-4788-b5d6-f17023c21623)
[test_46.zip](https://github.com/skhatiri/UAV-Testing-Competition/files/13652507/test_46.zip)

**Comment by skhatiri** [comment, marked as answer]
Such test scenarios, given the complexity of the obstacle placements, and the lack of (or difficulty in finding) a potential safe path to the destination are considered "infeasible". 
In the context of the competition, since we are particularly focusing on *safety aspects* of the flight, such test cases are invalid, as specifically stated [here](https://github.com/skhatiri/UAV-Testing-Competition/tree/master#uav-test-generators) and there is no value in reporting such test cases among your final generated tests:
> - The obstacle configurations are expected to keep the flight mission physically feasible.
    The test cases that make it impossible for the UAV to find its path (e.g., creating a long wall among the drone path) while there is no hard or soft fail are considered Invalid.

**Reply by dgumenyuk** [reply]
Thank you for the reply! Then it would be very helpful if you could formalize this requirement e.g. there should be at least one pair of obstacles with the distance more than 6 meters between them. How do you define the safe path?

**Reply by skhatiri** [reply]
Thanks for the feedback. 

I believe such a requirement is very dynamic and depends on the mission path, and the count, and the relative placement of the obstacles which makes it too hard to formulate. So far, the only way we can really know if a test is feasible is to run the simulation and see if the drone is able to find its path. Also, it could be very close to some **safety-failing** scenarios that we are interested in. 

Probably using the flight logs of such incomplete missions, or adding access to the runtime data (during the flight) could be more useful to detect them early and adapt the test generation process. We could consider these for the next year.
